### PR TITLE
[quant] aten::repeat work for quantized tensor

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40653 [quant][graphmode][fix] remove unsupported ops in the list
* **#40644 [quant] aten::repeat work for quantized tensor**
* #40624 [quant][graphmode][fix] cloning schema in insert_observers

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22268558](https://our.internmc.facebook.com/intern/diff/D22268558)

### CODE REVIEW

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -762,7 +762,12 @@ Tensor repeat(const Tensor& self, IntArrayRef repeats) {
 
   Tensor xtensor = self.expand(padded_size);
 
-  Tensor result = at::empty(target_size, self.options());
+  Tensor result;
+  if (self.is_quantized()) {
+    result = at::empty_quantized(target_size, self);
+  } else {
+    result = at::empty(target_size, self.options());
```
Comment (vkuzo) [inline aten/src/ATen/native/TensorShape.cpp:769]: does dispatching on tensor type not work for `at::empty`?

**Review by vkuzo — APPROVED** [review]

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -1099,6 +1099,11 @@
   variants: method
   device_guard: False
 
+- func: empty_quantized(int[] size, Tensor original) -> Tensor
```
Comment (supriyar) [inline aten/src/ATen/native/native_functions.yaml:1102]: The name seems a little non-intuitive to me. Maybe make it explicit that original has to be a quantized tensor.

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/TensorShape.cpp:769]: yeah, quantized parameters are not included in TensorOptions(self.options())

File: aten/src/ATen/native/native_functions.yaml (hunk 2 above)
Comment (jerryzh168) [reply aten/src/ATen/native/native_functions.yaml:1102]: sg

**Pull request closed without merging**